src/app.js

```diff
@@ -2,137 +2,243 @@
   const [chain, nodeName, nodeVersion] = await Promise.all([
     api.rpc.system.chain(),
     api.rpc.system.name(),
     api.rpc.system.version(),
   ]);
-  
+
   console.log(`You are connected to chain ${chain} using ${nodeName} v${nodeVersion}`);
 
+  return api;
+}
+
+const cennznet_transaction = {
+  transfer: (api, receiver, funds) =>
+    api.tx.genericAsset.transfer(16001, receiver, funds),
+  balance: (api, sender_address) =>
+    api.query.genericAsset.freeBalance(16001, sender_address)
+}
+
+const plug_transaction = {
+  transfer: (api, receiver, funds) =>
+    api.tx.balances.transfer(receiver, funds),
+  balance: (api, sender_address) =>
+    api.query.balances.freeBalance(sender_address)
+}
+
+async function setup(settings) {
+  // Command line delay used to wait for nodes to come up
+  await sleep(settings.startup_delay_ms);
+
+  let api;
+  let transaction;
+
+  if (settings.api ===  "plug") {
+    api = await getPlugApi(settings.address);
+    transaction = plug_transaction;
+  }
+  else {
+    api = await getCennznetApi(settings.address);
+    transaction = cennznet_transaction;
+  }
+
   // The test ends when the final block number = start + delta
   const start_block_number = await getFinalizedBlockNumber(api)
-  
+
   // Gather all required users
-  const keyring = testingPairs.default({ type: 'sr25519'});
+  const testKey = new Keyring({ type: 'sr25519'});
 
-  const required_users = 2*settings.transaction.timeout_ms/settings.transaction.period_ms;
-  const required_steves = Math.max(1, Math.floor(required_users-6));
+  const required_users = 2 * settings.transaction.timeout_ms / settings.transaction.period_ms;
+  const required_steves = Math.max(1, Math.floor(required_users));
 
-  const steve_keyring = new Keyring.Keyring({type: 'sr25519'});
+  const steve_keyring = new Keyring({type: 'sr25519'});
   const steves = createTheSteves(required_steves, steve_keyring);
-  await fundTheSteves(steves, api, keyring.alice, settings.transaction.timeout_ms);
-  
+  const aliceKeyring = testKey.addFromUri('//Alice');
+  const bobKeyring = testKey.addFromUri('//Bob');
+  const charlieKeyring = testKey.addFromUri('//Charlie');
+  const ferdieKeyring = testKey.addFromUri('//Ferdie');
+  const daveKeyring = testKey.addFromUri('//Dave');
+  const eveKeyring = testKey.addFromUri('//Eve');
+
+  if(settings.staking_validators > 0){
+    await addStaking.createStashAccounts(api, transaction);
+  }
+
+  if (settings.fund) {
+    await fundTheSteves(
+      steves,
+      api,
+      transaction,
+      [aliceKeyring, bobKeyring, charlieKeyring, ferdieKeyring, daveKeyring, eveKeyring],
+      settings.transaction.timeout_ms
+      );
+  }
+
   const keypair_selector = new selector.KeypairSelector(
-    [keyring.alice, keyring.bob, keyring.charlie, keyring.ferdie, keyring.dave, keyring.eve]
-    .concat(steves)
-    );
+    steves
+  );
 
-  const funds = 5;
+  const funds = 2000000000000;
```

Comment:
If we make this small, we can run the test longer without having to fund the steves again

Reply:
the tricky part is if the funds are too low, we can't see the change on our UI - this is eq to 2 DEV unit per tx. And to be fair because all the steves round-robin the fund so they shouldn't run out of fund even couple of them failing tx occasionally for a fair long time

Reply:
That isn't entirely true, sending funds is done by round robin, but receiving isn't (currently). So it is possible that over time some steves will have many more funds than others just by chance

Reply:
Kicked it down by four 0s, I want to keep it big enough so even from the UI interface we can see the balance different after awhile. It should sufficient for one steve at least spent 10K TXs, what do you think?

Reply:
Good compromise 👍